libjulia/backends/evm/EVMCodeTransform.cpp

```diff
@@ -51,10 +99,19 @@ void CodeTransform::operator()(VariableDeclaration const& _varDecl)
 		while (variablesLeft--)
 			m_assembly.appendConstant(u256(0));
 	}
-	for (auto const& variable: _varDecl.variables)
+	for (auto const& variable: _varDecl.variables | boost::adaptors::reversed)
 	{
 		auto& var = boost::get<Scope::Variable>(m_scope->identifiers.at(variable.name));
-		m_context->variableStackHeights[&var] = height++;
+		if (!m_context->unusedStackSlots.empty()) {
+			int slot = m_context->unusedStackSlots.back();
+			m_context->unusedStackSlots.pop_back();
+			m_assembly.appendInstruction(solidity::swapInstruction(height + numVariables - slot - 1));
```

Comment:
This should set the proper source location: `m_assembly.setSourceLocation(var.location)`.